.github/workflows/dependencies/clang.sh

```diff
@@ -1,45 +1,50 @@
 #!/usr/bin/env bash
 #
-# Copyright 2024 The WarpX Community
+# Copyright 2025 The WarpX Community
 #
 # License: BSD-3-Clause-LBNL
-# Authors: Luca Fedeli
 
 set -eu -o pipefail
 
-# `man apt.conf`:
-#   Number of retries to perform. If this is non-zero APT will retry
-#   failed files the given number of times.
+# `man apt.conf`: number of retries to perform (if non-zero,
+# APT will retry failed files the given number of times).
 echo 'Acquire::Retries "3";' | sudo tee /etc/apt/apt.conf.d/80-retries
 
-# This dependency file is currently used within a docker container,
-# which does not come (among others) with wget, xz-utils, curl, git,
-# ccache, and pkg-config pre-installed.
-sudo apt-get -qqq update
+sudo apt-get update
 sudo apt-get install -y \
     cmake               \
-    clang-17            \
-    clang-tidy-17       \
     libblas-dev         \
-    libc++-17-dev       \
     libboost-math-dev   \
     libfftw3-dev        \
     libfftw3-mpi-dev    \
     libhdf5-openmpi-dev \
     liblapack-dev       \
     libopenmpi-dev      \
-    libomp-17-dev       \
-    ninja-build         \
-    wget                \
-    xz-utils            \
-    curl                \
-    git                 \
-    ccache              \
-    pkg-config
+    ninja-build
 
-# Use clang 17
-export CXX=$(which clang++-17)
-export CC=$(which clang-17)
+# parse clang version number from command line
+version_number=${1}
```

Comment:
Saw too late.

You could make the script more robust by checking that the number of arguments are exactly one (or two?) (0 is the program/script name, 1 the first argument) and otherwise throw a helpful error message. In bash, the number of args is `$#`